sql/analyzer/aliases.go

```diff
@@ -116,7 +116,7 @@ func getTableAliases(n sql.Node, scope *Scope) (TableAliases, error) {
 			rt := getResolvedTable(node.Destination)
 			analysisErr = passAliases.add(rt, rt)
 			return false
-		case *plan.ResolvedTable, *plan.SubqueryAlias, *plan.ValueDerivedTable, *plan.TransformedNamedNode, *plan.RecursiveTable:
+		case *plan.ResolvedTable, *plan.SubqueryAlias, *plan.ValueDerivedTable, *plan.TransformedNamedNode, *plan.RecursiveTable, *plan.RecursiveCte:
```

Comment:
We're currently updating this list as we find more cases, which seems like an inefficient and error-prone approach. Should we add a `default` panic case here and identify all the possible cases? Then when we add more types that should be handled here, our tests will fail and force us to update this spot.

Reply:
Not sure if I agree that a default panic is in order, but this does speak to a missing interface.

Reply:
refactoring to match `sql.Nameable` and include exclusionary no-op cases where appropriate